api/routers/export.py

```diff
@@ -0,0 +1,169 @@
+import os
+import re
+import tempfile
+import logging
+
+from datetime import date, datetime
+from fastapi import APIRouter, HTTPException, BackgroundTasks
+from fastapi.responses import FileResponse
+from starlette import status
+from typing import List, Tuple
+from zipfile import ZipFile, ZIP_DEFLATED
+
+from api.models.export import ExportDTO, ExportDataType
+from api.utils import extract_config, clean_up_file
+from libs.metrics import FaceMaskUsageMetric, OccupancyMetric, SocialDistancingMetric
+
+logger = logging.getLogger(__name__)
+
+export_router = APIRouter()
+
+# Define the exports data types as constants
+ALL_DATA = ExportDataType.all_data
+RAW_DATA = ExportDataType.raw_data
+SOCIAL_DISTANCING = ExportDataType.social_distancing
+FACEMASK_USAGE = ExportDataType.facemask_usage
+OCCUPANCY = ExportDataType.occupancy
+
+
+def export_folder_into_zip(source_path, destination_path, zip_file, from_date, to_date):
+    if not os.path.exists(source_path):
+        return None
+    for filename in os.listdir(source_path):
+        if filename.endswith(".csv"):
+            if from_date and to_date:
+                # Date range was specified, only include files in that range
+                date_matches = re.findall(r"([0-9]{4}\-[0-9]{2}\-[0-9]{2})", filename)
+                if not date_matches:
+                    continue
+                file_date = datetime.strptime(date_matches[0], '%Y-%m-%d').date()
+                if not from_date <= file_date <= to_date:
+                    continue
+            zip_file.write(os.path.join(source_path, filename), arcname=os.path.join(destination_path, filename))
+
+
+def get_areas(export_info: ExportDTO) -> List[Tuple[str, str]]:
+    all_areas = extract_config("areas").values()
+    selected_areas = []
+    if export_info.all_areas:
+        selected_areas = all_areas
+    else:
+        selected_areas = [a for a in all_areas if a["Id"] in export_info.areas]
+        if len(selected_areas) != len(export_info.areas):
+            # Some of the selected areas don't exit
+            missing_areas = set(export_info.areas) - set([a["Id"]for a in selected_areas])
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"Areas with ids {missing_areas} don't exist."
+            )
+    if selected_areas:
+        return [(area["Id"], area["Name"]) for area in selected_areas]
+    return []
+
+
+def get_cameras(export_info: ExportDTO, areas: List[Tuple[str, str]]) -> List[Tuple[str, str]]:
```

Comment:
This function exports data from cameras in export_info and cameras in `areas` but it is not immediately obvious what it does. Maybe a short doctring could explain it. I see that we aren't adding them to other endpoints but it would still be good at least when the function is not completely obvious

Reply:
I include the docstring in the commit 31ec50682a5b52bc597a408b64508660880fbcc9